feat(pie-docs): add testing for accessibility page (#459)
justeattakeaway/pie: apps/pie-docs/src/accessibility/testing-for-accessibility.md

---
eleventyNavigation:
    key: Testing for accessibility
    parent: Accessibility
    order: 3
title: Testing for accessibility

> description: Find out more about how we test for accessibility and ensure ensure that our products meet accessibility standards.

Reviewer: how about another ensure :DDDD
![image](https://github.com/justeattakeaway/pie/assets/28605803/a5f0444f-d4d0-49fb-bee9-fb2786161eb7)

Reviewer 2: Lol! Thanks for noticing, that's been updated in Figma now!

Author: :D

> description: Find out more about how we test for accessibility and ensure ensure that our products meet accessibility standards.

Reviewer: ```suggestion
description: Find out more about how we test for accessibility and ensure that our products meet accessibility standards.
```

Reviewer 2: @FayeCarter Changed in design

---

## Why do we need to test for accessibility?

At JET we understand that accessibility is crucial when it comes to designing and building products. We are aiming for our products and services to meet at least **level AA** of the **Web Content Accessibility Guidelines 2.1 (WCAG)**. To achieve this, it's essential that you consider accessibility at the start of a project and run accessibility tests regularly throughout the design process.

We offer guidance on the different types of accessibility testing that should be used in your projects. We recommend a **mixture of automated, manual, and user testing  methods** to ensure that most accessibility issues can be detected.

> 

Reviewer 2: @FayeCarter Changed in design

{% contentPageImage {
    src:"../../../assets/img/accessibility/testing-for-accessibility/aa-wcag.svg",
    mobileSrc:"../../../assets/img/accessibility/testing-for-accessibility/aa-wcag_narrow.svg",
    caption: "Our products should meet the AA level of WCAG 2.1’s accessibility guidelines."
} %}

---

## Automated testing

> Automated testing tools can audit your product against the WCAG to ensure that you're following best practices that can be detected.

Reviewer 3: I'm guessing we've been given this content, but this sentence doesn't read very well (could get rid of `that can be detected`).

Reviewer 2: Rewrote this in Figma, should be easier to read now :)

> We estimate that **automated testing can detect a great amount of common WCAG issues**. Automated testing also gives your team more time to focus on the more complex accessibility tests that need to be completed manually.

Reviewer 3: Same here – I would right this as:

```suggestion
We estimate that **automated testing can detect a large number of common WCAG issues**. Automated testing also gives your team more time to focus on the more complex accessibility tests that need to be completed manually.
```

Reviewer 2: Agree with all suggestions, let's amend it :D

We recommend using tools such as Deque’s [axe-core](https://www.deque.com/axe/) for the web, [Google’s Toolbox for Accessibility for iOS](https://github.com/google/GTXiLib) (GTXiLib), or [Google’s Accessibility Test Framework for Android](https://github.com/google/Accessibility-Test-Framework-for-Android) (ATF) for automated testing.

{% notification {
  type: "information",
  message: "Please [reach out to us](/support/contact-us/) if you need help integrating these tools."
} %}

It’s important to note that whilst automated accessibility tests are a great step towards providing an accessible application, **they do not guarantee that your app is accessible to real users**.

> Some of the issues that automatic checks don’t usually cover are:

Reviewer: suggestion: 
```suggestion
Some of the issues that automated checks don’t usually cover are:
```

Reviewer 2: Updated in Figma now, @FayeCarter

Author: @jordifornellsJET

- Correct reading order of content
- Appropriate use of HTML
- Good keyboard navigation experience
- Useful image captions and alt text
- Cognitive accessibility

{% notification {
  type: "warning",
  message: "Automated tests should always be used in combination with a variety of manual test methods."
} %}

---

## Manual testing

For manual accessibility testing, we recommend referring to a comprehensive [accessibility checklist](https://www.a11yproject.com/checklist/) for web products to help you test for common accessibility barriers.

**The A11y Project checklist** provides an excellent resource that includes items such as checking for proper colour contrast, keyboard accessibility, and the correct use of descriptive text and alt attributes for images. This checklist can serve as a good foundation for testing the accessibility of your product.

{% notification {
  type: "information",

>   message: "You can use The A11y Project’s [accessibility checklist](https://www.a11yproject.com/checklist/) as a main resource to manual test your products."

Reviewer 3: ```suggestion
  message: "You can use The A11y Project’s [accessibility checklist](https://www.a11yproject.com/checklist/) as a main resource to manually test your products."
```

Author: @jordifornellsJET

Reviewer 2: Let's make this change (I'll update Figma)

        "**Form Accessibility**: We ensure that form elements, labels, and error messages are properly associated and announced.",
        "**ARIA Roles and Attributes**: We use [Accessible Rich Internet Applications](https://www.w3.org/WAI/standards-guidelines/aria/) (ARIA) roles and attributes to enhance the accessibility of dynamic and interactive elements. We use screen reader testing to verify that ARIA roles and attributes are correctly implemented and announced by the screen reader, providing additional context and functionality.",
        "**Multimedia Accessibility**: Screen reader testing also involves assessing the accessibility of multimedia elements, such as audio and video. We use it to ensure that appropriate alternative text or captions are provided for multimedia content."
    ]
} %}

---

## Testing using Applause

We have partnered with [Applause](https://www.applause.com/), a third-party crowdsourcing company, to help us with accessibility testing through a contract.

> Through this partnership, we can arrange **one-off specialised audits**, conducted by trained accessibility testers from Applause, for both new and existing features. By doing this, we can help teams **understand the state of accessibility in their current products** and how they can improve them to increase their level of accessibility.

Author: @jordifornellsJET

> Through this partnership, we can arrange **one-off specialised audits**, conducted by trained accessibility testers from Applause, for both new and existing features. By doing this, we can help teams **understand the state of accessibility in their current products** and how they can improve them to increase their level of accessibility.

Reviewer 2: @FayeCarter Changed in design

{% notification {
  type: "information",

>   message: "If your team would like to schedule and audit then [get in touch](/support/contact-us/) with us."

Reviewer 2: Change made in Figma. Thanks! :)

>   message: "If your team would like to schedule and audit then [get in touch](/support/contact-us/) with us."

Author: @jordifornellsJET

>   message: "If your team would like to schedule and audit then [get in touch](/support/contact-us/) with us."

Reviewer 3: ```suggestion
  message: "If your team would like to schedule an audit then [get in touch](/support/contact-us/) with us."
```

Reviewer 2: @FayeCarter I fixed this in previous comments, so it should be solved now in designs, thanks!
